core/launch.go

```diff
@@ -0,0 +1,225 @@
+package core
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"chain/core/accesstoken"
+	"chain/core/account"
+	"chain/core/asset"
+	"chain/core/config"
+	"chain/core/fetch"
+	"chain/core/generator"
+	"chain/core/leader"
+	"chain/core/pin"
+	"chain/core/query"
+	"chain/core/rpc"
+	"chain/core/txbuilder"
+	"chain/core/txdb"
+	"chain/core/txfeed"
+	"chain/database/pg"
+	"chain/log"
+	"chain/protocol"
+	"chain/protocol/bc"
+)
+
+const (
+	blockPeriod              = time.Second
+	expireReservationsPeriod = time.Second
+)
+
+// LaunchOption describes a runtime configuration option.
+type LaunchOption func(*API)
```

Comment:
I don't want to bikeshed these names too much, but just a little…

```
func Run
func RunUnconfigured
type Option
```

???

or perhaps RunOption instead of Option to keep them more strongly associated like you did here.

I'm also down with these names as-is.